packit/command_handler.py

```diff
@@ -110,6 +110,26 @@ def run_command(
 class SandcastleCommandHandler(CommandHandler):
     name = RunCommandType.sandcastle
 
+    def __init__(self, local_project: LocalProject, config: Config):
+        super().__init__(local_project=local_project, config=config)
+        self.local_project = local_project
+        self.config = config
+
+        # we import here so that packit does not depend on sandcastle (and thus python-kube)
+        from sandcastle.api import Sandcastle, MappedDir
+
+        self.mapped_dir = MappedDir(
+            local_dir=self.local_project.working_dir,
+            path=self.config.command_handler_work_dir,
+            with_interim_pvc=True,
+        )
+        self.sandcastle = Sandcastle(
+            image_reference=self.config.command_handler_image_reference,
+            k8s_namespace_name=self.config.command_handler_k8s_namespace,
+            mapped_dir=self.mapped_dir,
+        )
+        self.sandcastle.run()
+
     def run_command(
```

Comment:
I [see](https://github.com/packit/packit/blob/5dfeb65834f68d64c9d16a567ec0fae60efc0a1e/packit/base_git.py#L294) in `PackitRepositoryBase` that commands for actions are run one-by-one. Is this something we need to keep? Could we just merge all commands from an action into one and pass it to the sandbox to execute?

Reply:
How do you envision the merging into one would be implemented?

Reply:
Take all the commands, put it in a multiline string and pass it to Sandcastle, where it would end up in target script.

Reply:
We'd need a support for that in sandcastle since commands are invoked with [exec](https://ss64.com/bash/exec.html) so that return code is correctly propagated.

Your suggestion is absolutely good and we should do it - I just think it's out of scope of this PR because it's not an easy fix.

Reply:
https://github.com/packit/sandcastle/issues/107